src/simple_cache_client.rs

```diff
@@ -59,6 +59,13 @@ pub struct SimpleCacheClientBuilder {
     user_agent_name: String,
 }
 
+pub fn request_meta_data<T>(request: &mut tonic::Request<T>, cache_name: &str) {
+    request.metadata_mut().append(
+        "cache",
+        tonic::metadata::AsciiMetadataValue::try_from(cache_name).unwrap(),
+    );
+}
```

Comment:
Unwrap is an antipattern in SDK's. It causes user applications to crash for things that, really, we have no idea whether the user should crash over.

Consider plumbing the MomentoError up like
```
pub fn request_meta_data<T>(request: &mut tonic::Request<T>, cache_name: &str) -> Result<(), MomentoError> {
    Ok(
        tonic::metadata::AsciiMetadataValue::try_from(cache_name)
            .map(|value| {
                request.metadata_mut().append("cache", value);
                ()
            })
            .map_err(|e| {
                MomentoError::InvalidArgument(format!("Could not treat cache name as a header value: {}", e.to_string()))
            })?
    )
}
```

And when you call request_meta_data you call it like
```
request_meta_data(&mut request, cache_name)?;
```

so that when a bad value is encountered you bail upward from there.